deltachat_derive/src/lib.rs

```diff
@@ -8,36 +8,33 @@ use quote::quote;
 // data.  If this assumption is violated, compiler error will point to
 // generated code, which is not very user-friendly.
 
-#[proc_macro_derive(ToSql)]
-pub fn to_sql_derive(input: TokenStream) -> TokenStream {
+#[proc_macro_derive(Sqlx)]
+pub fn sqlx_derive(input: TokenStream) -> TokenStream {
     let ast: syn::DeriveInput = syn::parse(input).unwrap();
     let name = &ast.ident;
 
     let gen = quote! {
-        impl rusqlite::types::ToSql for #name {
-            fn to_sql(&self) -> rusqlite::Result<rusqlite::types::ToSqlOutput> {
-                let num = *self as i64;
-                let value = rusqlite::types::Value::Integer(num);
-                let output = rusqlite::types::ToSqlOutput::Owned(value);
-                std::result::Result::Ok(output)
+        impl sqlx::encode::Encode<sqlx::sqlite::Sqlite> for #name {
+            fn encode(&self, buf: &mut Vec<sqlx::sqlite::SqliteArgumentValue>) {
+                num_traits::ToPrimitive::to_i64(self).expect("invalid type").encode(buf)
```

Comment:
Rather unfortunate that errors are not allowed here.  Any idea why?

Reply:
Design issue in the library, it will be solved eventually.